### PR TITLE
Bugfix/Fix Diffie-Hellman bug - check if theirPublicKey is 0 or 1

### PR DESCRIPTION
**Pull request by GeorgeMcLellan**
The previous code `theirPublicKey < minValue` does not work for `JKBigInteger` I don't think. It led to the unit tests failing as the exception was always throws.

Instead, I used string comparison to just check if the value is "0" or "1", and wrote some unit tests to cover these scenarios.

https://mx51.atlassian.net/browse/INTG-495

### CODE REVIEW

File: Tests/Crypto/SPIDiffieHellmanTest.m [hunk 1]
```diff
@@ -32,7 +32,7 @@ - (void)testExchange {
     JKBigInteger *primeG = [[JKBigInteger alloc] initWithString:@"2"];
     
     JKBigInteger *privateKeyA = [SPIDiffieHellman randomPrivateKey:primeP];
-    JKBigInteger *privateKeyB = [SPIDiffieHellman randomPrivateKey:primeG];
+    JKBigInteger *privateKeyB = [SPIDiffieHellman randomPrivateKey:primeP];
```
Comment (GeorgeMcLellan) [inline Tests/Crypto/SPIDiffieHellmanTest.m:35]: I believe this was wrong, as primeG is only 2. This caused private & public keys to be <2 and the test to fail now that we have the check.

**Review by alistair-mx51 — DISMISSED** [review]
Great. LGTM. Thanks @GeorgeMcLellan 👍

File: Tests/Crypto/SPIDiffieHellmanTest.m (hunk 1 above)
Comment (alistair-mx51) [reply Tests/Crypto/SPIDiffieHellmanTest.m:35]: Both g and p are static. g (i.e. the generator) should always be 2. And p is the "prime modulus" which is always set to the same special large number. If it helps, further detail on this is on slide 6 of https://docs.google.com/presentation/d/1-IyRnjJoyjLTOcpWZS4928yHWbf6hX11wUZ0pUFFJh8/edit#slide=id.g22e8e426d90_0_39.

**Review by rossi-mx51 — APPROVED** [review]

File: Tests/Crypto/SPIDiffieHellmanTest.m (hunk 1 above)
Comment (GeorgeMcLellan) [reply Tests/Crypto/SPIDiffieHellmanTest.m:35]: Yep I think thats all being done currently, just the generation of a random private key was resulting in 1 for privateKeyB. The parameter we pass in to the function randomPrivateKey() is a max integer value, and the function generates a random value between 0 and that value. So I think we were just passing in the wrong value to this function

**Pull request merged**